vignettes/dp-histogram.Rmd

```diff
@@ -0,0 +1,247 @@
+
+```{r}
+# When entering bins, enter the minimum of the range, the upper bound of each bin, and the maximum of the range
+incomeBins <- c(0,50000, 100000, 200000, 300000, 400000, 750000)
+
+# So in this example, we have 6 bins:
+# 1. (min, 500000]
+# 2. (50000, 100000]
+# 3. (100000, 200000]
+# 4. (200000, 300000]
+# 5. (300000, 400000]
+# 6. (400000, max]
+# where min = the minimum of the dataset = 0
+# and max = the maximum of the dataset = 750000
+
+numeric_histogram <- dpHistogram$new(var.type='numeric', variable='income', n=10000, epsilon=0.1, 
+                                     rng=c(0,750000), n.bins=6, bins=incomeBins)
+numeric_histogram$release(PUMS5extract10000)
+print(numeric_histogram$result)
+```
+
+Note: entering bins is completely optional and should only be done if you have prior knowledge of the data.
+
+&nbsp;
+
+To calculate the histogram of a logical variable, input a logical vector into `variable` and update `var.type` to 'logical'. Also add a boolean flag for `impute` to indicate if you would like empty values to be imputed from the existing data. (If `impute = TRUE`, then there will only be two bins in the histogram: 0 and 1. If `impute = FALSE`, then there will be three bins in the histogram: 0, 1, and NA.)
+
+```{r}
+logical_histogram <- dpHistogram$new(var.type='logical', variable='married', n=10000, epsilon=0.1, impute = TRUE)
+logical_histogram$release(PUMS5extract10000)
+print(logical_histogram$result)
+```
+
+In the example above, we set the `impute` parameter equal to `TRUE`, meaning that any `NA` (empty) values would be randomly assigned 0 or 1 so no values in the data set are empty, and there is no `NA` bucket in the histogram. By default `impute = FALSE`, so if we do not specify a value for `impute`, then there will be an `NA` bucket in the histogram release:
+
+```{r}
+logical_histogram <- dpHistogram$new(var.type='logical', variable='married', n=10000, epsilon=0.1)
+logical_histogram$release(PUMS5extract10000)
+print(logical_histogram$result)
+```
+
+For logical variables, it is never necessary to enter a vector of bins, you just need to specify whether to impute empty values or not. It is also not necessary to enter a range for logical variables, because the range of a logical variable is known to be c(0,1).
+
+&nbsp;
+
+To calculate the histogram of a categorical (or 'character') variable, input a character variable name into `variable` and update `var.type` to 'character'.
+
+```{r}
+character_histogram <- dpHistogram(var.type='character', variable='tobgp', n=88, epsilon=1.5, delta=10^-4)
```

Comment:
Note here epsilon > 1 but above we say we require epsilon to be < 1. So maybe we should just say epsilon should be  "generally < 1"